core/src/main/scala/dagr/core/execsystem/TaskManager.scala

```diff
@@ -523,7 +528,29 @@ class TaskManager(taskManagerResources: TaskManagerResources = TaskManagerDefaul
     )
   }
 
+  /** Refreshes the top-like interface */
+  private class TerminalThread(terminal: TopLikeStatusReporter) extends Runnable {
+    override def run(): Unit = {
+      try {
+        if (0 < taskTop) {
+          // skip over any current output
+          print(CursorMovement.clearScreen)
+          // keep printing
+          while (true) {
+            terminal.refresh(print)
+            Thread.sleep((taskTop * 1000).toInt)
+          }
+        }
+      } catch {
+        case e: InterruptedException => // ignore
+      }
+    }
+  }
+
   override def runToCompletion(failFast: Boolean): BiMap[Task, TaskExecutionInfo] = {
+    val terminalThread = if (0 < taskTop) Some(new Thread(new TerminalThread(this))) else None
+    terminalThread.foreach(_.start())
```

Comment:
A couple of thoughts on the handling of the thread.

1. It would good to call `setDaemon(true)` on the terminalThread, so that if something goes wrong and the main thread exits without interrupting the terminalThread, the VM can still exit
2. I think I would be inclined to implement a "tidier" shutdown mechanism for the thread.  I'd suggest adding an AtomicBoolean to TerminalThread, which you can reach in from the outside and flip instead of using `Thread.interrupt()`, and that the thread checks each iteration, and exits if requested.